mystery-methods.rb

```diff
@@ -14,6 +14,16 @@ def mystery2(n)
   end
 end
 
+def mystery2_new(n)
+  if n.abs < 10
+    return n
+  elsif n > 0
+    return (n%10) + mystery2_new(n/10)
+  else
+    return -(n.abs%10) + mystery2_new((n/10.0).ceil)
+  end
+end
+
```

Comment:
This looks like it works, but it's a little confusing to read. Here's another solution that's maybe a little more readable:

```ruby 
def mystery2(n)
  if n < 0
    return -1 * mystery2(-n)
  elsif n < 10
    return n
  else
    return (n%10) + mystery2(n/10)
  end
end
```